proto/xmtpv4/envelopes/envelopes.proto

```diff
@@ -60,3 +60,45 @@ message OriginatorEnvelope {
     BlockchainProof blockchain_proof = 3;
   }
 }
+
+enum Misbehavior {
+  MISBEHAVIOR_UNSPECIFIED = 0;
+  MISBEHAVIOR_UNRESPONSIVE_NODE = 1;
+  MISBEHAVIOR_SLOW_NODE = 2;
+  MISBEHAVIOR_FAILED_REQUEST = 3;
+  MISBEHAVIOR_OUT_OF_ORDER = 4;
+  MISBEHAVIOR_DUPLICATE_SEQUENCE_ID = 5;
+  MISBEHAVIOR_CAUSAL_ORDERING = 6;
+  MISBEHAVIOR_INVALID_PAYLOAD = 7;
+  MISBEHAVIOR_BLOCKCHAIN_INCONSISTENCY = 8;
+}
+
+message LivenessFailure {
+  uint32 response_time_ns = 1;
+  SubscribeEnvelopesRequest subscribe = 2;
+  QueryEnvelopesRequest query = 3;
+  PublishPayerEnvelopesRequest publish = 4;
+}
+
+message SafetyFailure {
+  repeated OriginatorEnvelope envelopes = 1;
+}
+
+message UnsignedMisbehaviorReport {
+  uint64 reporter_time_ns = 1;
+  uint32 misbehaving_node_id = 2;
+  Misbehavior type = 3;
+  LivenessFailure liveness = 4;
+  SafetyFailure safety = 5;
+  // Nodes must verify this field is false for client-submitted reports
+  bool submitted_by_node = 6;
+}
+
+message MisbehaviorReport {
+  // Server time when the report was stored. Used only for querying reports.
+  // This field is not signed.
+  uint64 server_time_ns = 1;
```

Comment:
We need this separately to the reporter time (which may have client skew), so that the query endpoint can query all payloads larger than a certain timestamp. Could have used a sequence ID or a timestamp, decided to use a timestamp so that it's simpler, while also avoiding confusion with cursors in the main replication protocol